### ISSUE TITLE
Enable Binary logging support 

### ISSUE THREAD

**Issue by jsturtevant** [containerd/runwasi#630]
Containerd has pluggable logging which includes binary logging: https://github.com/containerd/containerd/blob/c7f7fa5390f761b83c3a69cd5709a8d04d912969/core/runtime/v2/README.md#logging

Currently we support fifo/file/npipe

**Comment by squillace** [comment]
well, per https://github.com/containerd/containerd/blob/c7f7fa5390f761b83c3a69cd5709a8d04d912969/core/runtime/v2/README.md#logging they are one of those supported on Windows, at a minimum:

Shims may support pluggable logging via STDIO URIs. Current supported schemes for logging are:

fifo - Linux
binary - Linux & Windows
file - Linux & Windows
npipe - Windows

**Comment by squillace** [comment]
the main benefit of binary logging is that you can do highly structured snapshots and various other customized things to not only compress but treat the log as a kind of database. So, for example, systemd and mysql (in fact, almost every db or db service) uses binary logging. Size is not the **main** concern, but "structured responsiveness". 

outside of that, you can just compress the text log if size is your only concern (though ongoing size is typically smaller with binary).

In this case, we have a Windows customer who needs this to make runwasi a "drop-in" replacement for the current shim.

**Comment by jsturtevant** [comment]
From https://github.com/containerd/containerd/issues/6639:

> Example policies that a custom log plugin might enforce include:
>Log data is pre-processed to remove potentially sensitive information (e.g. via https://github.com/stripe/unilog)
>Pre-processed log data is not allowed to be written to persistent disk.
>Log lines may be persisted to remote storage at different sampling levels depending on content.

learned a little bit more about how binary loggers should work: they are expected to be called, and piping I/O in to the custom logger.  An example the logger is in https://github.com/containerd/containerd/pull/3085

**Comment by Mossaka** [comment]
Yeah sounds good. Perhaps this issue belongs to rust-extensions repo, though?
